components/base/wheeled/kinematics.go

```diff
@@ -0,0 +1,110 @@
+package wheeled
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/golang/geo/r3"
+	"github.com/pkg/errors"
+	"go.viam.com/rdk/components/base"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/services/slam"
+	"go.viam.com/rdk/spatialmath"
+)
+
+type kinematicWheeledBase struct {
+	*wheeledBase
+	slam  slam.Service
+	model referenceframe.Model
+}
+
+func WrapWithKinematics(base *wheeledBase, slam slam.Service) (base.KinematicBase, error) {
+	var err error
+	kwb := &kinematicWheeledBase{
+		wheeledBase: base,
+		slam:        slam,
+	}
+	kwb.model, err = kwb.buildModel()
+	if err != nil {
+		return nil, err
+	}
+	return kwb, err
+}
+
+func (kwb *kinematicWheeledBase) ModelFrame() referenceframe.Model {
+	return kwb.model
+}
+
+func (kwb *kinematicWheeledBase) CurrentInputs(ctx context.Context) ([]referenceframe.Input, error) {
+	// TODO(RSDK-2311): complete the implementation
+	return []referenceframe.Input{}, errors.New("not implemented yet")
+}
+
+func (kwb *kinematicWheeledBase) GoToInputs(ctx context.Context, goal []referenceframe.Input) error {
+	// TODO(RSDK-2311): complete the implementation
+	return errors.New("not implemented yet")
+}
+
+func (kwb *kinematicWheeledBase) buildModel() (referenceframe.Model, error) {
+	if kwb.collisionGeometry == nil {
+		return nil, errors.New("cannot create model for base with no collision geometry")
+	}
+
+	// get the limits of the SLAM map to set as the extents of the frame
+	data, err := slam.GetPointCloudMapFull(context.Background(), kwb.slam, "")
+	if err != nil {
+		return nil, err
+	}
+	pcd, err := pointcloud.ReadPCD(bytes.NewReader(data))
+	if err != nil {
+		return nil, err
+	}
+	dims := pcd.MetaData()
+
+	// build the model - SLAM convention is that the XZ plane is the ground plane
+	frame2D, err := referenceframe.NewMobile2DFrame(
+		kwb.collisionGeometry.Label(),
+		[]referenceframe.Limit{{Min: dims.MinX, Max: dims.MaxX}, {Min: dims.MinZ, Max: dims.MaxZ}},
+		kwb.collisionGeometry)
+	if err != nil {
+		return nil, err
+	}
+	model := referenceframe.NewSimpleModel(kwb.name)
+	model.OrdTransforms = []referenceframe.Frame{frame2D}
+	return model, nil
+}
+
+func collisionGeometry(cfg config.Component) (spatialmath.Geometry, error) {
```

Comment:
Just realized something else that's an issue here.

When we created the Remote Geometries scope, it was explicitly decided that geometries added to components with nonzero DoF would be placed prior to the transform of the component in question; think a geometry on an arm representing the table it's attached to; not something that gets waved around on the end of the EE.

So the geometry specified here would get attached to the Offset frame.

Reply:
We're kind of getting around this because we're building the model from scratch and it exists separate from the frame system.  That being said, yeah thats a little hacky and possibly inconsistent.  Is there a fix in mind?

Reply:
The offset geometry will exist in addition to the model. That needs to be actively disabled somehow if you want to use this, otherwise the base will leave a big sphere behind on the floor wherever it was when it turned on.

Reply:
Bases being added to the frame system was entirely descoped.  Does this address your problem?

Reply:
Ah; that makes it better but doesn't entirely fix it.

The base geometry, if it is in the RDK config, *will* get added to the frame system regardless of whether the base uses the frame system. This is fine for the base. But this is not fine for anything else that does not use the base but does generally consume the frame system, as there will now be a large sphere ~somewhere~ to contend with.

This is likely less of an issue now though.

Reply:
No there won't be a large sphere because the KWB will only be created for the lifespan of a single motion call.  Normal bases do not implement ModelFramer and therefore will not add it.  There will however be whatever geometry the user specified to the base.  But I think thats something we can live with without special cases

Reply:
Right- not a sphere, sorry. Just the actual geometry being parsed.

Reply:
That should at least be documented somewhere.

Reply:
Ok I will add a comment to the model function noting how it doesn't actually get added to the frame system.  Is that what you mean? Or some other type of documentation